### PR TITLE
Metrics: Replace <string, string> labels with attributes

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
It would be great to benchmark this to see the performance impact.

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (jsuereth) [inline opentelemetry/proto/metrics/v1/metrics.proto:355]: Nit:  Should we call out error scenarios and what to do with competing types in same-label-key across different Metrics, or just but a TODO to call that out in the DataModel Spec?

**Comment by Oberon00** [comment]
Since according to our maturity definition for beta (which metrics are) only allows one breaking change every 3 months, do we want to downgrade to alpha? I don't think we want to wait till end of June with releasing this?

**Review by jsuereth — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@bogdandrutu can you also link for the record to a specification issue or PR where the decision was made to move from strings to full attributes?

File: opentelemetry/proto/metrics/v1/metrics.proto
Comment (jmacd) [reply opentelemetry/proto/metrics/v1/metrics.proto:355]: I am comfortable with a TODO to call this out in the data model spec. As it stands, this file could use a complete rewrite of its comments, but I'd rather discuss and debate what those comments should really say in the data model document, and then copy sections of the data model back into the .proto at the end.

**Review by jmacd — APPROVED** [review]

**Comment by jsuereth** [comment]
@tigrannajaryan Here's the recorded decision for this: https://github.com/open-telemetry/opentelemetry-specification/issues/1113#issuecomment-800525450

**Comment by tigrannajaryan** [comment]
> @tigrannajaryan Here's the recorded decision for this: [open-telemetry/opentelemetry-specification#1113 (comment)](https://github.com/open-telemetry/opentelemetry-specification/issues/1113#issuecomment-800525450)

@jsuereth thanks for recording the decision, this is very helpful.

**Comment by jsuereth** [comment]
Please add to description: Fixes #1113

**Review by carlosalberto — APPROVED** [review]

**Comment by hdost** [comment]
> Since according to our maturity definition for beta (which metrics are) only allows one breaking change every 3 months, do we want to downgrade to alpha? I don't think we want to wait till end of June with releasing this?

This is why I had brought up #275 I'm sure what the rush being getting 0.8 was if guarantees aren't being guaranteed then they don't mean anything. But also agree i wouldn't have even considered it Beta yet based on the discussions.

File: CHANGELOG.md
Comment (jsuereth) [inline CHANGELOG.md:18]: Discussed in Data Model SiG:  This isn't a breaking change because the deprecated fields are still included in the protocol.   Once removed, that would be breaking.

**Review by reyang — APPROVED** [review]
LGTM.

**Comment by jmacd** [comment]
I would like to know from @bogdandrutu and @tigrannajaryan how they think we should decide to merge or not merge this PR.

**Comment by tigrannajaryan** [comment]
> I would like to know from @bogdandrutu and @tigrannajaryan how they think we should decide to merge or not merge this PR.

My understanding is that Metrics SIG decided that metrics labels should become attributes. I trust you have considered all aspects and made this decision, so I have no objections.

My only ask would be that after merging this PR and any other final changes planned, before making a release to benchmark the proto and do an optimization pass to see if we can improve the proto performance wise, while maintaining the same semantics.

**Pull request merged**
